junebug/tests/helpers.py

```diff
@@ -182,10 +182,21 @@ def start_server(self):
         port = reactor.listenTCP(
             0, Site(self.api.app.resource()),
             interface='127.0.0.1')
-        self.addCleanup(port.stopListening)
+        self.service._port = port
+        self.addCleanup(self.stop_server)
         addr = port.getHost()
         self.url = "http://%s:%s" % (addr.host, addr.port)
 
+    @inlineCallbacks
+    def stop_server(self):
+        yield self.service.stopService()
+        for service in self.service.services:
+            yield service.stopService()
+        self.service.services = []
+        for service in self.service.namedServices.values():
+            yield service.stopService()
+        self.service.namedServices = {}
```

Comment:
Would using `self.service.removeService()` on each child service maybe simplify this? It seems to handle stopping the service and removing it from `service.services` and `service.namedServices`.